clrenv/lazy_env.py

```diff
@@ -198,22 +237,19 @@ def _apply_functions(d, recursive=False):
                     logger.warning(f"[{socket.gethostname()}] Offline, using placeholder value for {parameter_name}.")
                     value = OFFLINE_VALUE
                 else:
-                    try:
-                        value = _get_ssm_client().get_parameter(
-                            Name=parameter_name,
-                            WithDecryption=True
-                        )['Parameter']['Value']
-                    except EndpointConnectionError as e:
-                        raise RuntimeError(
-                            "clrenv could not connect to AWS to fetch parameters. If you're developing locally, try setting the offline environment variable (CLRENV_OFFLINE_DEV) to use placeholder values.")
-                    except _get_ssm_client().exceptions.ParameterNotFound as e:
-                        raise RuntimeError(
-                            f"Could not find {parameter_name} in Parameter Store.")
+                    # Don't fetch the parameters here, but use a placeholder value.
+                    # Save a reference to the current environment and key.
+                    # In _resolve_parameters(), this placeholder will be replaced with the actual value.
+                    _parameter_store_key_cache[parameter_name] = (new, key)
```

Comment:
the env `new` is going to be the same here for all the keys. Seems like we should be able to pass in `new` into `resolve_parameters` as well

Reply:
It's not, it's going to be a sub-environment of the main one.